File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/core/data/SwlHeader.js
/**
 * @return {SwlHeader}
 * @constructor
 */
function SwlHeader() {
    /**
     * 时间戳
     * @type {String}
     */
    this.timestamp = null;
    /**
     * 一次性消息
     * @type {String}
     */
    this.nonce = null;
    /**
     * 随机秘钥
     * @type {String}
     */
    this.randomKey = null;
    /**
     * 签名
     * @type {String}
     */
    this.sign = null;
    /**
     * 数字签名
     * @type {String}
     */
    this.digital = null;

    /**
     * 服务端非对称签名
     * @type {String}
     */
    this.remoteAsymSign = null;

    /**
     * 客户端非对称签名
     * @type {String}
     */
    this.localAsymSign = null;
}

/**
 *
 * @param src {SwlHeader}
 * @param dst {SwlHeader}
 * @return {SwlHeader}
 */
SwlHeader.copy = function (src, dst) {
    dst.timestamp=src.timestamp;
    dst.nonce = src.nonce;
    dst.randomKey = src.randomKey;
    dst.sign = src.sign;
    dst.digital = src.digital;
    dst.localAsymSign = src.localAsymSign;
    dst.remoteAsymSign = src.remoteAsymSign;
    return dst;
}

export default SwlHeader

File: i2f-jdk/i2f-swl/src/main/java/i2f/swl/data/SwlHeader.java
package i2f.swl.data;

import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Ice2Faith
 * @date 2024/7/10 9:41
 * @desc
 */
@Data
@NoArgsConstructor
public class SwlHeader {
    /**
     * 时间戳
     */
    public String timestamp;
    /**
     * 一次性消息
     */
    public String nonce;
    /**
     * 随机秘钥
     */
    public String randomKey;
    /**
     * 签名
     */
    public String sign;
    /**
     * 数字签名
     */
    public String digital;
    /**
     * 证书ID
     */
    public String certId;


    public static SwlHeader copy(SwlHeader src, SwlHeader dst) {
        dst.setTimestamp(src.getTimestamp());
        dst.setNonce(src.getNonce());
        dst.setRandomKey(src.getRandomKey());
        dst.setSign(src.getSign());
        dst.setDigital(src.getDigital());
        dst.setCertId(src.getCertId());
        return dst;
    }
}
